framework/include/image_data.hpp

```diff
@@ -0,0 +1,282 @@
+#pragma once
+#include <gvk.hpp>
+
+namespace gvk
+{
+	class image_data_t;
+	class image_data_implementor;
+
+	using image_data = avk::owning_resource<image_data_t>;
```

Comment:
That's ok, I understood your previous comments in the way that all data types should have the owning_resource functionality, but I agree it is of not much use for image_data since the data itself will be loaded unto the heap anyways.